core/src/main/java/hudson/model/Run.java

```diff
@@ -1566,34 +1569,51 @@ public synchronized void deleteArtifacts() throws IOException {
     public void delete() throws IOException {
         File rootDir = getRootDir();
         if (!rootDir.isDirectory()) {
-            throw new IOException(this + ": " + rootDir + " looks to have already been deleted; siblings: " + Arrays.toString(project.getBuildDir().list()));
+            //No root directory found to delete. Somebody seems to have nuked
+            //it externally. Logging a warning before dropping the build
+            LOGGER.warning(String.format(
```

Comment:
This pattern requires the string formatting even if the message is not printed. `LOGGER.log(Level.WARNING, <pattern>, new Obcject[] {args})` is a better pattern. Not an issue for WARNINGs tho, they are printed in all common cases